lib/note-content-editor.tsx

```diff
@@ -150,8 +152,96 @@ class NoteContentEditor extends Component<Props> {
         }
       }, 400);
     }
+
+    if (this.editor && prevProps.searchQuery !== this.props.searchQuery) {
+      this.setDecorators();
+    }
   }
 
+  setDecorators = () => {
+    const searchMatches = this.searchMatches();
+    const titleDecoration = this.getTitleDecoration();
+    if (!searchMatches.length && !titleDecoration) {
+      return;
+    }
+    if (titleDecoration) {
+      searchMatches.push(titleDecoration);
+    }
+    decorations = this.editor.deltaDecorations(decorations, [...searchMatches]);
```

Comment:
we have some syntax confusion. on one hand we're _spreading_ something that's already a list and on the other we're mutating a list that's later spread.

we can resolve this mismatch in one of two ways: make everything immutable, or make everything mutable. you know my preference would be towards the immutable…

```ts
setDecorators = () => {
     const searchMatches = this.searchMatches() ?? [];
     const titleDecoration = this.getTitleDecoration() ?? [];

     this.decorations = this.editor.deltaDecorations(
          this.decorations,
          [...searchMatches, ...titleDecoration]
     );
}
```

in this snippet I've made three changes:
 - I've used `this.decorations` assuming that we've [made `decorations` a class instance member](https://github.com/Automattic/simplenote-electron/pull/2292/files#r475878496) of the `NoteContentEditor` vs. a module-global shared by all instances
 - in the case that one of the components has no matches we're defaulting to `[]` so we can safely spread them
 - we're _always_ updating the decorations. if we early-abort when there are no new decorations then we'll continue to highlight the old ranges when we are supposed to stop highlighting altogether